### PR TITLE
Harden UnifiedMessenger.java

### PR DESCRIPTION
**Pull request by DanVanAtta**
- Add precondition checks to fail sooner on error conditions
- Add final keyword
- Remove unused method

### CODE REVIEW

File: src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java [hunk 1]
```diff
@@ -10,6 +10,9 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
 import games.strategy.engine.message.ConnectionLostException;
 import games.strategy.engine.message.HubInvocationResults;
 import games.strategy.engine.message.HubInvoke;
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java]: The `games.strategy.util.ThreadUtil` import is no longer used.

File: src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java [hunk 2]
```diff
@@ -178,12 +182,14 @@ public INode getLocalNode() {
   public Object getImplementor(final String name) {
     synchronized (m_endPointMutex) {
       final EndPoint endPoint = m_localEndPoints.get(name);
+      Preconditions.checkNotNull(endPoint, "local endpoints: " +
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java]: Checkstyle: "'+' should be on a new line."

File: src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java [hunk 3]
```diff
@@ -320,11 +302,18 @@ public void messageReceived(final Serializable msg, final INode from) {
       // all
       synchronized (m_pendingLock) {
         m_results.put(methodId, results.results);
-        m_pendingInvocations.remove(methodId).countDown();
+        final CountDownLatch latch = m_pendingInvocations.remove(methodId);
+        Preconditions.checkNotNull(latch, "method id: " + methodId + ", was not present in pending invocations: " + m_pendingInvocations + ", unified messenger addr: " + super.toString());
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java]: Checkstyle: "Line is longer than 120 characters (found 188)."

**Comment by DanVanAtta** [comment]
Were checkstyle violations the only thing you would have wanted fixed on this one @ssoloff , or good to go when they are fixed?

**Pull request merged**